### ISSUE TITLE
Validation losses increase while training losses decrease

### ISSUE THREAD

**Issue by hadim** [matterport/Mask_RCNN#766]
Here are losses plots with training in blue and validation in red.

![image831](https://user-images.githubusercontent.com/528003/42524540-ced34390-843e-11e8-87be-6d8db980ebeb.png)

My validation dataset represents 15% of the entire dataset. I also tried with 25%. I am using the coco dataset to init the parameters of the model and `resnet50` as a backbone.

My dataset only contains one label. Here are two examples of images coming from the dataset: 

![image831](https://user-images.githubusercontent.com/528003/42524722-5a8b6fd4-843f-11e8-99fc-c0f2fa58444c.png)

I am training on `heads` with a LR of 0.001. I also tried to train on `4+` with a LR of `0.0001` with the same result.

Did someone also encounter this issue of increasing validation losses? It looks like the network is overfitting. Any suggestions are welcome.

**Comment by zungam** [comment]
How big is your dataset?

I would recomend to use alot of data augmentation (google "imgaug", its allready implented in mask_rcnn).

I also see that your plots are like waves / zig zaging. Try using shuffle on validation and training set.

**Comment by hadim** [comment]
Thank you for your inputs @zungam.

Yes I use augmentation during training:

```python
augmentation = iaa.SomeOf((0, 2), [
        iaa.Fliplr(0.5),
        iaa.Flipud(0.5),
        iaa.OneOf([iaa.Affine(rotate=90),
                   iaa.Affine(rotate=180),
                   iaa.Affine(rotate=270)]),
        iaa.Multiply((0.8, 1.5)),
        iaa.GaussianBlur(sigma=(0.0, 2.0))])
```

I shuffle the training dataset but not the validation dataset (as it's a common practice usually).

As for the spikes on the training curve (blue), it happens when I restart the training process... Do you see the same? Something must be wrong because when you resume training from the last weights, the losses should be the same as the previous epoch right?

**Comment by wdyxwzyh** [comment]
seems something happens when u resume training. The META thing of ckpt didnt include the status of random seed,so it results in a difference. See here: https://github.com/endernewton/tf-faster-rcnn.git
And search this question in readme.md. But the author dont know how to do either,so am i.Maybe u can output something.

**Comment by bhushaan** [comment]
@hadim : Did you get around this issue? I also have just one label and validation loss graph is a zigzag.

**Comment by hadim** [comment]
Not really...

**Comment by JiayingLi** [comment]
Hi,  can anyone tell me where do you get these plots from? Thanks

**Comment by banafsh89** [comment]
These are from the tensor board. You give the log dir to tensor board and plot it.

**Comment by s33k3rs** [comment]
@hadim can you tell me, how to show training graph/blue line ? Thanks
![cad](https://user-images.githubusercontent.com/11786565/56155427-11a9c900-5fe5-11e9-9217-21a9fb78909c.jpg)

**Comment by 1044886782** [comment]
> Here are losses plots with training in blue and validation in red.
> 
> ![image831](https://user-images.githubusercontent.com/528003/42524540-ced34390-843e-11e8-87be-6d8db980ebeb.png)
> 
> My validation dataset represents 15% of the entire dataset. I also tried with 25%. I am using the coco dataset to init the parameters of the model and `resnet50` as a backbone.
> 
> My dataset only contains one label. Here are two examples of images coming from the dataset:
> 
> ![image831](https://user-images.githubusercontent.com/528003/42524722-5a8b6fd4-843f-11e8-99fc-c0f2fa58444c.png)
> 
> I am training on `heads` with a LR of 0.001. I also tried to train on `4+` with a LR of `0.0001` with the same result.
> 
> Did someone also encounter this issue of increasing validation losses? It looks like the network is overfitting. Any suggestions are welcome.

could you tell me how to draw those loss curves ? I'm a newbie...

**Comment by tomklaver1995** [comment]
Can anyone help me to get the same plots as hadim? So both the validation as the training losses in one graph. Not all seperate like s33k3rs

**Comment by banafsh89** [comment]
> Can anyone help me to get the same plots as hadim? So both the validation as the training losses in one graph. Not all seperate like s33k3rs

You can download the data file from Tensorflow plots (validation and training losses) and then from those data plot what ever you want yourself.  In the tensorflow there is an option that you can download the raw data into your computer.
